pints/_mcmc/__init__.py

```diff
@@ -774,16 +830,30 @@ def run(self):
             if self._log_to_screen:
                 print(halt_message)
 
-        # Store generated chains in memory
-        if self._chains_in_memory:
-            self._samples = samples
-
         # Store evaluations in memory
         if self._evaluations_in_memory:
             self._evaluations = evaluations
 
+        # Inverse transform to model space if transform is provided
```

Comment:
I think I'd prefer to do this on-the-fly, instead of after we're done! But maybe you can convince me otherwise :D

Reply:
Interesting one, I thought I would like the run keep in the search space to avoid any potential issue. If one decided to use the stored samples to look at the next tell() point in the future or such like. Not 100% convinced that one way is the best.